src/openforms/config/tests/test_global_configuration.py

```diff
@@ -165,6 +165,56 @@ def test_virus_scan_enabled_cant_connect(self):
             "Cannot connect to ClamAV: Cannot connect!",
         )
 
+    @override_settings(LANGUAGE_CODE="en")
+    def test_svg_sanitation_favicon(self):
+        bad_svg_content = (
+            '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 50 50">'
+            '<circle cx="25" cy="25" r="25" fill="green" onclick="alert(\'forget about me?\')" />'
+            "<script>//<![CDATA["
+            'alert("I am malicious >:)")'
+            "//]]></script>"
+            "<g>"
+            '<rect class="btn" x="0" y="0" width="10" height="10" fill="red" onload="alert(\'click!\')" />'
+            "</g>"
+            "</svg>"
+        )
+        sanitized_svg_content = (
+            '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 50 50">'
+            '<circle cx="25" cy="25" r="25" fill="green"></circle>'
+            "//"
+            'alert("I am malicious &gt;:)")'
+            "//"
+            "<g>"
+            '<rect x="0" y="0" width="10" height="10" fill="red"></rect>'
+            "</g>"
+            "</svg>"
+        )
+
+        config = GlobalConfiguration.get_solo()
+
+        with self.subTest(part="admin config"):
+            url = reverse("admin:config_globalconfiguration_change", args=(1,))
+            change_page = self.app.get(url)
+
+            upload = Upload(
+                "bad_svg.svg", bad_svg_content.encode("utf-8"), "image/svg+xml"
+            )
+
+            form = change_page.forms["globalconfiguration_form"]
+            form["favicon"] = upload
+            _ensure_arrayfields(form, config=config)
+            response = form.submit()
+
+            self.assertEqual(response.status_code, 302)
+            config.refresh_from_db()
+            self.assertEqual(config.favicon, "logo/bad_svg.svg")
+
+        with self.subTest(part="assert favicon sanitized"):
+            with config.favicon.file.open("rb") as favicon_file:
+                # Assert that the logo is completely sanitized
+                decoded_logo = favicon_file.read().decode("utf-8")
```

Comment:
```suggestion
            with config.favicon.file.open("r") as favicon_file:
                # Assert that the logo is completely sanitized
                decoded_logo = favicon_file.read()
```
should have the same result, but opening the file in text mode rather than binary allows you to skip the decoding step

Reply:
This results into a miss-match in amount of characters, when using the `assertEqual(...)`. This is because we then would compare a byte-string to a regular string. Maybe if the expected value is also defined as a byte string, this would work..
I'll take a look

Reply:
Yeah, that does the trick 👍